### PR TITLE
adwsteamgtk: init at 0.6.9

### LINKED ISSUE #228958: Package request: adwsteamgtk
**Project description**

A simple GTK wrapper that installs and updates the [Adwaita for Steam](https://github.com/tkashkin/Adwaita-for-Steam) skin.

**Metadata**

* homepage URL: https://github.com/Foldex/AdwSteamGtk
* source URL: https://github.com/Foldex/AdwSteamGtk
* license: gpl3
* platforms: linux

### PR DESCRIPTION
**Pull request by reedrw**
## Description of changes

Add [AdwSteamGtk](https://github.com/Foldex/AdwSteamGtk), an installer for the [Adwaita-for-Steam](https://github.com/tkashkin/Adwaita-for-Steam) Steam skin.

Fixes https://github.com/NixOS/nixpkgs/issues/228958

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by bb010g — APPROVED** [review]
Concerns expressed out-of-band were resolved. LGTM.

File: pkgs/by-name/ad/adwsteamgtk/package.nix [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    desktop-file-utils
+    meson
+    ninja
+    wrapGAppsHook
+  ];
+
+  propagatedBuildInputs = [
+    python3Packages.packaging
+    python3Packages.pygobject3
+  ];
```
Comment (tomodachi94) [inline pkgs/by-name/ad/adwsteamgtk/package.nix:42]: Overall lgtm, just a minor suggestion so we stay DRY (don't repeat yourself).
```suggestion
  propagatedBuildInputs = with python3Packages; [
    packaging
    pygobject3
  ];
```

**Review by tomodachi94 — CHANGES_REQUESTED** [review]

File: pkgs/by-name/ad/adwsteamgtk/package.nix [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+  meta = {
+    description = "A simple Gtk wrapper for Adwaita-for-Steam";
+    homepage = "https://github.com/Foldex/AdwSteamGtk";
+    license = lib.licenses.gpl3Only;
```
Comment (Aleksanaa) [inline pkgs/by-name/ad/adwsteamgtk/package.nix:47]:
```suggestion
    license = lib.licenses.gpl3Plus;
```

Should be okay to use gpl3Plus, as https://github.com/Foldex/AdwSteamGtk/blob/b73387001004f91ebd3bc479034370ef22a8a979/data/io.github.Foldex.AdwSteamGtk.appdata.xml.in#L4

File: pkgs/by-name/ad/adwsteamgtk/package.nix [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    libadwaita
+    libportal-gtk4
+  ];
+
+  nativeBuildInputs = [
+    blueprint-compiler
+    desktop-file-utils
+    meson
+    ninja
+    wrapGAppsHook
```
Comment (Aleksanaa) [inline pkgs/by-name/ad/adwsteamgtk/package.nix:36]:
```suggestion
    wrapGAppsHook4
```

**Review by tomodachi94 — APPROVED** [review]
Lgtm, thank you for this package!

**Review by Aleksanaa — APPROVED** [review]
lgtm

**Review by jnsgruk — APPROVED** [review]

**Comment by Kreyren** [comment]
Was this designed to be declarative or is this meant to be installed imperatively? (would a simple user service do the trick?)

**Pull request merged**